lib/metasploit/framework/data_service/remote/http/remote_loot_data_service.rb

```diff
@@ -6,7 +6,16 @@ module RemoteLootDataService
   LOOT_PATH = '/api/1/msf/loot'
 
   def loot(opts = {})
-    json_to_open_struct_object(self.get_data(LOOT_PATH, opts), [])
+    # TODO: Add an option to toggle whether the file data is returned or not
+    loots = json_to_open_struct_object(self.get_data(LOOT_PATH, opts), [])
+    # Save a local copy of the file
+    loots.each do |loot|
+      if loot.data
+        local_path = File.join(Msf::Config.loot_directory, File.basename(loot.path))
```

Comment:
As we verify that `loot.data` is not nil before proceeding should we also verify that `loot.path` is not nil or an empty string before using it?

Reply:
So this probably isn't the best behavior, but I didn't add that check for the sake of protective programming. My goal with checking for `loot.data` was for the case where we had a loot object with no data sent over. Currently we always send it, but I added a TODO about adding something to the request about to only send the loot metadata and not the contents of the file.